src/Orchard.Cms.Web/Modules/Orchard.OpenId/Recipes/OpenIdSettingsStep.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+using Orchard.Admin;
+using Orchard.Recipes.Models;
+using Orchard.Recipes.Services;
+using Orchard.OpenId.Services;
+using System.Security.Cryptography.X509Certificates;
+using static Orchard.OpenId.Settings.OpenIdSettings;
+using System.Collections.Generic;
+
+namespace Orchard.OpenId.Recipes
+{
+    public class OpenIdSettingsStep : RecipeExecutionStep
+    {
+        private readonly IOpenIdService _openIdService;
+        
+        public OpenIdSettingsStep(IOpenIdService openIdService,
+            ILoggerFactory logger,
+            IStringLocalizer<OpenIdSettingsStep> localizer) : base(logger, localizer)
+        {
+            _openIdService = openIdService;
+        }
+
+        public override string Name
+        {
+            get { return "OpenIdSettings"; }
+        }
+
+        public override async Task ExecuteAsync(RecipeExecutionContext recipeContext)
+        {
+            var model = recipeContext.RecipeStep.Step.ToObject<OpenIdSettingsStepModel>();
+
+            var openIdSettings = await _openIdService.GetOpenIdSettingsAsync();
+            openIdSettings.TestingModeEnabled = model.TestingModeEnabled;
+            openIdSettings.DefaultTokenFormat = model.DefaultTokenFormat;
+            openIdSettings.Audience = string.Join(",",model.Audience);
```

Comment:
Is there a particular reason `OpenIdSettings.Audience(s)` can't be an array? I suppose the underlying settings stuff doesn't support that?